rasterio/plot.py

```diff
@@ -126,10 +128,11 @@ def show(source, with_bounds=True, contour=False, contour_label_kws=None,
 
     if adjust and arr.ndim >= 3:
         # Adjust each band by the min/max so it will plot as RGB.
-        arr = reshape_as_raster(arr)
+        arr = reshape_as_raster(arr).astype("float64")
```

Comment:
This is the heart of it. We have to ensure that imshow receives a float array when `adjust=True`.